src/test/java/cmpt276/pg6/realtorest/controllers/MainControllerTest.java

```diff
@@ -32,4 +32,79 @@ static void setUp() {
     void testShowHomePage() throws Exception {
         this.mockMvc.perform(MockMvcRequestBuilders.get("/")).andExpect(MockMvcResultMatchers.status().isOk()).andExpect(MockMvcResultMatchers.view().name("home"));
     }
+
+    @Test
+    void testShowLoginPage() throws Exception {
+        this.mockMvc.perform(MockMvcRequestBuilders.get("/login")).andExpect(MockMvcResultMatchers.status().isOk()).andExpect(MockMvcResultMatchers.view().name("users/login"));
+    }
+    @Test
+    void testShowRegisterPage() throws Exception {
+        this.mockMvc.perform(MockMvcRequestBuilders.get("/register")).andExpect(MockMvcResultMatchers.status().isOk()).andExpect(MockMvcResultMatchers.view().name("users/register"));
+    }
+    @Test
+    void testShowPropertyListingsPage() throws Exception {
+        this.mockMvc.perform(MockMvcRequestBuilders.get("/property-listings"))
+        .andExpect(MockMvcResultMatchers.status().isOk())
+        .andExpect(MockMvcResultMatchers.view().name("propertyListings"));
+    }
+
+    @Test
+    void testUserLogin() throws Exception {
+        this.mockMvc.perform(MockMvcRequestBuilders.post("/login")
+        .param("email", "email@example.com")
+        .param("password", "password12"))
+        .andExpect(MockMvcResultMatchers.status().isAccepted())
```

Comment:
This test is not passing since this user doesn't exist in the database. And no mock data is created for it.

Reply:
Seems like even if the user exists in the actual database, the test still won't pass. You will need to add some mock data before testing this.